From 49935e70b07d1e0ae6853ca1bd6d587dda562a71 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Wed, 20 Mar 2019 09:16:28 +0000
Subject: [PATCH] rhbz#1690645 null deref of pEntry when GetCurEntry return
 null

in GetFocusRect at SvTreeList::GetDepth

code in since....

commit 76c549eb01dcb7b5bf28a271ce00e386f3d388ba
Author: Steve Yin <steve_y@apache.org>
Date:   Fri Nov 29 13:03:27 2013 +0000

    Integrate branch of IAccessible2

Change-Id: Ida6cb934c94037c861c7d5da005226f32599b5fc
---
 sw/source/uibase/utlui/content.cxx | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/sw/source/uibase/utlui/content.cxx b/sw/source/uibase/utlui/content.cxx
index 1bb69c686343..01021f7129e1 100644
--- a/sw/source/uibase/utlui/content.cxx
+++ b/sw/source/uibase/utlui/content.cxx
@@ -2850,15 +2850,14 @@ void SwContentTree::KeyInput(const KeyEvent& rEvent)
     //and realize multi-selection .
     else if(aCode.GetCode() == KEY_SPACE && 0 == aCode.GetModifier())
     {
-
         SvTreeListEntry* pEntry = GetCurEntry();
-        if( GetChildCount( pEntry ) == 0 )
-            m_bIsKeySpace = true;
-        Point tempPoint = GetEntryPosition( pEntry );//Change from "GetEntryPos" to "GetEntryPosition" for acc migration
-        m_aOldRectangle = GetFocusRect( pEntry,tempPoint.Y() );
-
         if(pEntry)
         {
+            if( GetChildCount( pEntry ) == 0 )
+                m_bIsKeySpace = true;
+            Point tempPoint = GetEntryPosition( pEntry );//Change from "GetEntryPos" to "GetEntryPosition" for acc migration
+            m_aOldRectangle = GetFocusRect(pEntry, tempPoint.Y());
+
             if (State::HIDDEN != m_eState)
             {
                 if (State::CONSTANT == m_eState)
-- 
2.20.1

